coax/reward_tracing/_nstep_verbose.py

```diff
@@ -0,0 +1,73 @@
+# ------------------------------------------------------------------------------------------------ #
+# MIT License                                                                                      #
+#                                                                                                  #
+# Copyright (c) 2020, Microsoft Corporation                                                        #
+#                                                                                                  #
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software    #
+# and associated documentation files (the "Software"), to deal in the Software without             #
+# restriction, including without limitation the rights to use, copy, modify, merge, publish,       #
+# distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the    #
+# Software is furnished to do so, subject to the following conditions:                             #
+#                                                                                                  #
+# The above copyright notice and this permission notice shall be included in all copies or         #
+# substantial portions of the Software.                                                            #
+#                                                                                                  #
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING    #
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND       #
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,     #
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,   #
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.          #
+# ------------------------------------------------------------------------------------------------ #
+
+from itertools import islice
+import jax.numpy as jnp
+
+from .._base.errors import InsufficientCacheError
+from ._nstep import NStep
+from ._transition import TransitionBatch
+
+__all__ = (
+    'NStep',
+)
+
+
+class NStepVerbose(NStep):
```

Comment:
We could integrate this into the normal `NStep` tracer but I am not sure whether that's the way to go. We would need to add an argmument to it.

Reply:
Yeah maybe in this case it might be better to add an argument to the existing `NStep` class.

Also, please could you add the intermediate actions and rewards in there too? This would make v-trace easier in the future.

Reply:
I moved the `extra_info` part to the `NStep` tracer. It is now recording all possible data for future use.